Harvest.Web/ClientApp/src/Projects/AdhocProject.tsx

```diff
@@ -0,0 +1,240 @@
+import React, { useCallback, useContext, useEffect, useState } from "react";
+import { useHistory, useParams } from "react-router-dom";
+import toast from "react-hot-toast";
+import {
+  Activity,
+  Expense,
+  AdhocProjectModel,
+  Rate,
+  WorkItemImpl,
+  ExpenseQueryParams,
+  Project,
+} from "../types";
+import { ActivityForm } from "../Quotes/ActivityForm";
+import { Button } from "reactstrap";
+
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faPlus } from "@fortawesome/free-solid-svg-icons";
+import { usePromiseNotification } from "../Util/Notifications";
+import AppContext from "../Shared/AppContext";
+
+import {
+  useOrCreateValidationContext,
+  ValidationProvider,
+} from "use-input-validator";
+import { workItemSchema } from "../schemas";
+import { checkValidity } from "../Util/ValidationHelpers";
+import * as yup from "yup";
+import { useQuery } from "../Shared/UseQuery";
+import { useIsMounted } from "../Shared/UseIsMounted";
+import { validatorOptions } from "../constants";
+import { authenticatedFetch } from "../Util/Api";
+import { convertCamelCase } from "../Util/StringFormatting";
+
+interface RouteParams {
+  projectId?: string;
+}
+
+const getDefaultActivity = (id: number) => ({
+  id,
+  name: "Generic Activity",
+  total: 0,
+  year: 0,
+  adjustment: 0,
+  workItems: [
+    new WorkItemImpl(id, 1, "Labor"),
+    new WorkItemImpl(id, 2, "Equipment"),
+    new WorkItemImpl(id, 3, "Other"),
+  ],
+});
+
+export const AdhocProject = () => {
+  const history = useHistory();
+
+  const [rates, setRates] = useState<Rate[]>([]);
+  const [inputErrors, setInputErrors] = useState<string[]>([]);
+  const context = useOrCreateValidationContext(validatorOptions);
+
+  // activities are groups of expenses
+  const [activities, setActivities] = useState<Activity[]>([
+    getDefaultActivity(1),
+  ]);
+
+  const [notification, setNotification] = usePromiseNotification();
+
+  const getIsMounted = useIsMounted();
+
+  useEffect(() => {
+    // get rates so we can load up all expense types and info
+    const cb = async () => {
+      const response = await authenticatedFetch("/api/Rate/Active");
+
+      if (response.ok) {
+        const rates: Rate[] = await response.json();
+
+        getIsMounted() && setRates(rates);
+
+        // create default activity
```

Comment:
?

Reply:
@srkirkland  It is in the expense container:
https://github.com/ucdavis/Harvest/blob/main/Harvest.Web/ClientApp/src/Expenses/ExpenseEntryContainer.tsx#L95